src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildWrapper.java

```diff
@@ -549,16 +564,12 @@ private static String getCurrentHostName() {
     /**
      * @return the port to be used
      */
-    private int getPort() {
+    private int getPort(Map<String, String> envVars) {
         if (StringUtils.isNotBlank(seleniumPort) && !seleniumPort.equals("0")) {
             Matcher matcher = ENVIRONMENT_VARIABLE_PATTERN.matcher(seleniumPort);
             if (matcher.matches()) {
                 String variableName = matcher.group(1);
-                String value = System.getenv(variableName);
-                if (value == null) {
-                    value = "0";
-                }
-                return Integer.parseInt(value);
+                return MapUtils.getInteger(envVars, variableName, 0);
```

Comment:
System variables from master are not being propagated to the build's environment => it may break existing use-cases. Both sources should be checked